File: leetcode/Python/4sum.py
class Solution:
    def fourSum(self, nums: List[int], target: int) -> List[List[int]]:
        s, t = set(), 0        
        nums.sort()
        for i in range(len(nums)-1, 2, -1):
            for j in range(i-1,1,-1):
                l, r = 0, j-1
                while l<r:
                    t=nums[l]+nums[r]+nums[j]+nums[i]
                    if t == target:
                        s.add((nums[l],nums[r],nums[j],nums[i]))
                        l+=1
                    elif t<target:
                        l+=1
                    else:
                        r-=1
        return list(s)

File: leetcode/C++/4sum.cpp
class Solution {
public:
    vector<vector<int>> fourSum(vector<int>& nums, int target) {
        set<vector<int>> s;
        int t,l,r;
        sort(nums.begin(),nums.end());
        for(int i=nums.size()-1; i>=3; --i){
            for(int j=i-1; j>=2; --j){
                l=0; r=j-1;
                while(l<r){
                    t=nums[l]+nums[r]+nums[j]+nums[i];
                    if(t==target) {
                        s.insert({nums[l],nums[r],nums[j],nums[i]}); 
                        ++l;
                    }
                    else if(t<target) ++l;
                    else --r;
                }
            }
        }
        return vector<vector<int>>(s.begin(), s.end());
    }
};